Documentation/workflow/UnitTests.md

```diff
@@ -0,0 +1,267 @@
+# Unit Tests
+
+
+## Tools
+
+1. [NUnit](https://github.com/nunit/nunit)
+2. [XUnit](https://github.com/xunit/xunit)
+
+## Project Test Types
+
+1. Build Tests
+
+    There are a number of `Build` related tests which test various aspects of the `.NET Android` build system. This includes not only MSBuild Task tests but build integration tests. Since the SDK is consumed from MSBuild, almost all of the unit tests are MSBuild related.
+
+2. Device Tests
+
+    Because `.NET Android` has to work on devices, we have a number of
+    "Integration" tests which check to make sure a final app will function as expected on an Emulator or Device. Our CI system will
+    test against an Emulator. However the system will pick up the first attached device, so developers can test one physical hardware
+    if needed.
+
+## Test Count
+
+| Test | Count |
+| :---------: | ----: |
+| Test | 0 |
+| Test | 100 |
+
+
+## Running Tests
+
+Running tests in the IDE is currently not supported.
+
+After building the repo you can make use of the `dotnet-local` script
+to run unit tests against the locally build SDK. The `dotnet-local` script is a wrapper around the custom `dotnet` installation which the
+build downloads and installs in `bin/Debug/dotnet` or `bin/Release/dotnet` (depending on your configuration).
+
+### MacOS/Linux
+
+To run ALL the build tests run.
+
+`dotnet-local.sh test bin/TestDebug/net7.0/Xamarin.Android.Build.Tests.dll --filter=Category!=DotNetIgnore`
```

Comment:
An example using `--filter "Name~Whatever"` would also be good, a'la `Documentation/workflow/DevelopmentTips.md`.  (Repetition is good, occasionally.)